### PR TITLE
Refactor TextEditViewController to use separate KeyboardViewController

### PR DESCRIPTION
**Pull request by jsmorgan42**
Refactoring the TextEditViewController to only be responsible for the top bar which is now not part of a collection view, meaning that the dismiss button & confirm edit button are no longer collection view cells but rather gazeable buttons. To display the keyboard, it embeds a KeyboardViewController that is responsible for the keyboard & suggestions. 

This KeyboardViewController should be used in the future as a component to refactor the keyboard within the main home screen as well.

### CODE REVIEW

File: Vocable/Features/Keyboard/KeyboardViewController.swift [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    }
+    
+    private func setupCollectionView() {
+        collectionView.delaysContentTouches = false
+        collectionView.isScrollEnabled = false
+        
+        collectionView.register(UINib(nibName: "KeyboardKeyCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "KeyboardKeyCollectionViewCell")
+        collectionView.register(UINib(nibName: "SuggestionCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "SuggestionCollectionViewCell")
+        collectionView.register(UINib(nibName: "FunctionKeyboardKeyCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "FunctionKeyboardKeyCollectionViewCell")
+        collectionView.register(UINib(nibName: "SpeakFunctionKeyboardKeyCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "SpeakFunctionKeyboardKeyCollectionViewCell")
```
Comment (stevefosterwta) [inline Vocable/Features/Keyboard/KeyboardViewController.swift]: Should these cells use the `.reuseIdentifier` for the reuseIdentifier (`KeyboardKeyCollectionViewCell.reuseIdentifier`) in place of a string?

File: Vocable/Features/Settings/EditCategories/EditCategories.storyboard [hunk 2]
```diff
@@ -96,7 +96,7 @@
                                 </constraints>
                                 <edgeInsets key="layoutMargins" top="0.0" left="0.0" bottom="0.0" right="0.0"/>
                             </view>
-                            <containerView opaque="NO" contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="ZPM-Ab-mlI">
+                            <containerView opaque="NO" contentMode="scaleToFill" ambiguous="YES" translatesAutoresizingMaskIntoConstraints="NO" id="ZPM-Ab-mlI">
```
Comment (stevefosterwta) [inline Vocable/Features/Settings/EditCategories/EditCategories.storyboard]: Is there a layout issue in the storyboard?

Comment (jsmorgan42) [reply Vocable/Features/Settings/EditCategories/EditCategories.storyboard]: Whoops, I think I have a few unintentional storyboard changes. I can go ahead & revert those changes

File: Vocable/Features/Keyboard/KeyboardViewController.swift (hunk 1 above)
Comment (jsmorgan42) [reply Vocable/Features/Keyboard/KeyboardViewController.swift]: Sure I think that makes more sense

**Review by Clstroud — APPROVED** [review]
💯 💯 💯

**Comment by stechkev** [comment]
"X" button and "check" button don't respond to headtracking taps. this is on the edit my sayings keyboard

**Review by stevefosterwta — APPROVED** [review]
🏁 👍

**Pull request merged**